### ISSUE TITLE
Kernel Errors cause SocketSession to hang

### ISSUE THREAD

**Issue by indig** [polynote/polynote#28]
I'm not sure how, but it looks like whenever we generate a Kernel Error we somehow lose connectivity between the UI and the backend. 

Reproduce:
- start a fresh polynote
- generate a Kernel Error
   - one easy way to do this is to try to make a new notebook with the same name as one that already exists. 
- try to do other stuff like running cells
   - it won't work!
- reload the page
- try to do stuff again
   - it works!

**Comment by jeremyrsmith** [comment]
I don't think I can repro this anymore? Can you?

**Comment by indig** [comment]
I can still reproduce:
- Run ServerApp
- navigate to http://localhost:8192/
- try to create a notebook that already exists. This causes an error: 
```
[ForkJoinPool-1-worker-15] ERROR polynote.server.SocketSession - FileAlreadyExistsException: foo.ipynb
polynote.kernel.RuntimeError: FileAlreadyExistsException: foo.ipynb
```
- you will be unable to to other things (open new notebooks, run cells, etc) until you reload the page. 
- after reloading then everything works normally

**Comment by indig** [comment]
The problem seems to be that when an error is thrown in a stream the stream terminates, and `handleErrorWith` doesn't prevent that, it just lets you do something with the error other than raising it. 

Unfortunately I wasn't able to find a way to handle a thrown exception without terminating the stream. I assume there must be a way to do this though...

**Comment by indig** [comment]
Note that the changes in #54 will make the reproduction described above obsolete, you would need to generate the error some other way.

**Comment by faisalzs** [comment]
We should fix this in 0.1.0

**Comment by jeremyrsmith** [comment]
I think this should be fixed... @indig is right about stream terminations not being preventable by `Stream#handleErrorWith`, but we've changed most of the `Stream.eval[IO, _]` calls to do `IO#handleErrorWith` on the `IO` where possible. Also, in places where the stream can be "restarted", `Stream#handleErrorWith` gets passed a function that will restart the stream, which does prevent termination.

Reopen this ticket if able to hang the session!
